src/structured_config.py

```diff
@@ -78,3 +80,22 @@ def ttl_minutes_validator(cls, value: str) -> Optional[int]:
         if int_value > 0:
             return int_value
         raise ValueError("Value out of range.")
+
+    @validator("logs_ttl")
+    @classmethod
+    def logs_ttl_validator(cls, value: str) -> Optional[int]:
+        """Check validity of `logs-ttl` fields.
+
+        Args:
+            value: logs-ttl value
+
+        Returns:
+            int_value: integer for logs-ttl configuration
+
+        Raises:
+            ValueError: in the case when the value is out of range
+        """
+        int_value = int(value)
```

Comment:
question (non-blocking): Can this ever get a `ValueError` for non-sensical input? Do we just let it raise an error?

Reply:
If the user sets a non-integer input on an integer config value, the charm will raise an error by default. This validator is to further extend this validation to ensure the value is within range.